### ISSUE TITLE
Feature: Notifications of important changes

### ISSUE THREAD

**Issue by kule** [mozilla/server-side-tls#44]
I love this generator however one thing that concerns me about having a cipher list is the what happens when new ones are added/others removed or there are important changes to the recommended setup?

Just wondering if there's something that could be put in place to notify users of important changes? e.g. perhaps just a simple mailchimp/sendgrid subscription for important changes only?

**Comment by gene1wood** [comment]
As this tool that generates web server config files consumes the ciphersuite list recommendations [published on the wiki page](https://wiki.mozilla.org/Security/Server_Side_TLS#Recommended_configurations) I'd say that any notification method for changes should originate from the wiki (as opposed to this tool).

@jvehent Any thoughts on how to communicate changes in the recommended ciphersuites? I'd say we could recommend people use the built-in `watchlist` function in MediaWiki but then _any_ changes on the entire page would trigger a notification (instead of just the ciphersuites).

We could publish the ciphersuites in a machine readable form and then try to get the wiki page to include them (It doesn't look like wiki.mozilla.org allows includes).

**Comment by jvehent** [comment]
> We could publish the ciphersuites in a machine readable form

I think that would be a valuable solution, maybe by simply hosting them on mozilla.github.io ?

I'm not keen on running a mailing list. It's extra work and we don't really have time for this. But we do announce changes on twitter so there's that...

**Comment by gene1wood** [comment]
@jvehent as the primary maintainer of the ciphersuite list, would you be open to maintaining the list in two places, one being the wiki in english and two being a machine readable location? I ask because I can't find a way to make the wiki consume something machine readable.

If so, I can publish the current ciphersuites machine readable (possibly on github) and work from that. This would give an easy way for people to "watch" if/when the ciphersuite recommendations change.

**Comment by jvehent** [comment]
Yep, that's fine. What about adding a json to this repo that's served on the static site for automated retrieval?

``` json
{
    "modern": {
        "ciphersuite": "ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-DSS-AES128-GCM-SHA256:kEDH+AESGCM:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-DSS-AES128-SHA256:DHE-RSA-AES256-SHA256:DHE-DSS-AES256-SHA:DHE-RSA-AES256-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:AES:CAMELLIA:DES-CBC3-SHA:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!MD5:!PSK:!aECDH:!EDH-DSS-DES-CBC3-SHA:!EDH-RSA-DES-CBC3-SHA:!KRB5-DES-CBC3-SHA",
        "protocols": [
            "tlsv1",
            "tlsv1.1",
            "tlsv1.2"
        ],
        "rsa_keysize": 2048,
        "dhe_keysize": 1024,
        "curves": [
            "prime256v1",
            "secp384r1",
            "secp521r1"
        ],
        "signature_algorithm": "sha256",
        "hsts": "max-age=15724800"
    },
    "intermediate": {
        "ciphersuite": "ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-DSS-AES128-GCM-SHA256:kEDH+AESGCM:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-DSS-AES128-SHA256:DHE-RSA-AES256-SHA256:DHE-DSS-AES256-SHA:DHE-RSA-AES256-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:AES:CAMELLIA:DES-CBC3-SHA:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!MD5:!PSK:!aECDH:!EDH-DSS-DES-CBC3-SHA:!EDH-RSA-DES-CBC3-SHA:!KRB5-DES-CBC3-SHA",
        "protocols": [
            "tlsv1",
            "tlsv1.1",
            "tlsv1.2"
        ],
        "rsa_keysize": 2048,
        "dhe_keysize": 1024,
        "curves": [
            "prime256v1",
            "secp384r1",
            "secp521r1"
        ],
        "signature_algorithm": "sha256",
        "hsts": "max-age=15724800"
    },
    "old": {
        "ciphersuite": "ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-DSS-AES128-GCM-SHA256:kEDH+AESGCM:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-DSS-AES128-SHA256:DHE-RSA-AES256-SHA256:DHE-DSS-AES256-SHA:DHE-RSA-AES256-SHA:ECDHE-RSA-DES-CBC3-SHA:ECDHE-ECDSA-DES-CBC3-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:AES:DES-CBC3-SHA:HIGH:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!MD5:!PSK:!aECDH:!EDH-DSS-DES-CBC3-SHA:!EDH-RSA-DES-CBC3-SHA:!KRB5-DES-CBC3-SHA",
        "protocols": [
            "sslv3",
            "tlsv1",
            "tlsv1.1",
            "tlsv1.2"
        ],
        "rsa_keysize": 2048,
        "dhe_keysize": 1024,
        "curves": [
            "prime256v1",
            "secp384r1",
            "secp521r1"
        ],
        "signature_algorithm": "sha1",
        "hsts": "max-age=15724800"
    }
}
```

**Comment by gene1wood** [comment]
Deal. Looks great!

**Comment by jvehent** [comment]
Note that the JSON version of the guidelines is being addressed in #97 . We don't have any plan for email notifications at the time, but interested consumers will be able to poll the json file regularly to check the version.
I'm closing this issue.
